Lombiq.Hosting.Tenants.EmailQuotaManagement/Services/EmailQuotaService.cs

```diff
@@ -58,60 +90,53 @@ public async Task<EmailQuota> GetCurrentQuotaAsync()
         currentQuota = new EmailQuota
         {
             // Need to set default value otherwise the database might complain about being 01/01/0001 out of range.
-            LastReminder = _clock.UtcNow.AddMonths(-1),
+            LastReminderUtc = _clock.UtcNow.AddMonths(-1),
         };
         _session.Save(currentQuota);
 
         return currentQuota;
     }
 
-    public void IncreaseQuota(EmailQuota emailQuota)
+    public void IncreaseEmailUsage(EmailQuota emailQuota)
     {
-        emailQuota.CurrentEmailQuotaCount++;
+        emailQuota.CurrentEmailUsageCount++;
         _session.Save(emailQuota);
     }
 
-    public void SaveQuotaReminder(EmailQuota emailQuota)
+    public void SetQuotaOnEmailReminder(EmailQuota emailQuota)
     {
-        emailQuota.LastReminder = _clock.UtcNow;
-        emailQuota.LastReminderPercentage = CurrentUsagePercentage(emailQuota);
+        emailQuota.LastReminderUtc = _clock.UtcNow;
+        emailQuota.LastReminderPercentage = emailQuota.CurrentUsagePercentage(GetEmailQuotaPerMonth());
         _session.Save(emailQuota);
     }
 
-    public bool ShouldSendReminderEmail(EmailQuota emailQuota, int? currentPercentage = null)
+    public bool ShouldSendReminderEmail(EmailQuota emailQuota, int currentUsagePercentage)
     {
-        currentPercentage ??= CurrentUsagePercentage(emailQuota);
-        if (currentPercentage < 80)
+        if (currentUsagePercentage < 80)
         {
             return false;
         }
 
-        var isSameMonth = IsSameMonth(_clock.UtcNow, emailQuota.LastReminder);
+        var isSameMonth = IsSameMonth(_clock.UtcNow, emailQuota.LastReminderUtc);
 
         if (!isSameMonth)
         {
             return true;
         }
 
-        switch (emailQuota.LastReminderPercentage)
-        {
-            case >= 80 when currentPercentage < 90:
-            case >= 90 when currentPercentage < 100:
-            case >= 100:
-                return false;
-            default:
-                return true;
-        }
+        return !((emailQuota.LastReminderPercentage >= 80 && currentUsagePercentage < 90) ||
+            (emailQuota.LastReminderPercentage >= 90 && currentUsagePercentage < 100) ||
+            emailQuota.LastReminderPercentage >= 100);
```

Comment:
Isn't this the same? 

```suggestion
        return  emailQuota.LastReminderPercentage < 90);
```

Reply:
I don't see how that ensures that we only send a reminder once in the 80%-90% and in the 90%-100% usage.

Reply:
Yeah, nevermind.